### PR TITLE
October Doc Cleanup based on Github issues

### PR DESCRIPTION
**Pull request by lkasso**
### Context
Handles Github issues and a few internal issues in sections that have been recently updated in the Gradle User Manual.

This is a documentation change ONLY.

### Fixes:
- Address https://github.com/gradle/gradle/issues/26659
- Fix https://github.com/gradle/gradle/issues/26814
- Fix https://github.com/gradle/gradle/issues/26861
- Fix https://github.com/gradle/gradle/issues/26750
- Fix version of Gradle Plugin Portal in docs for @runningcode 
- Update the description of [task_configuration_avoidance_migration_guidelines](https://docs.gradle.org/8.4/userguide/task_configuration_avoidance.html#sec:task_configuration_avoidance_migration_guidelines) requested anonymously
- Fix the broken link of [`ProjectLayout`](https://docs.gradle.org/8.4/userguide/upgrading_version_8.html#project_builddir) requested anonymously
- Update the file description of [Tutorial Part 4 step 3](https://docs.gradle.org/8.4/userguide/part4_gradle_plugins.html)
- Update the build file referenced in [Tutorial Part 3 step 1](https://docs.gradle.org/8.4/userguide/part3_gradle_dep_man.html) and add a command to [Tutorial Part 3 step 6](https://docs.gradle.org/8.4/userguide/part3_gradle_dep_man.html)
- Update where the code block should be placed in code in [Tutorial Part 4 step 3](https://docs.gradle.org/8.4/userguide/part3_gradle_dep_man.html)
- Fix bad Windows invocation of Gradle Wrapper in [Tutorial Part 1 step 3](https://docs.gradle.org/8.4/userguide/part1_gradle_init.html)

### Reviewers:
Under development.

### CODE REVIEW

**Review by runningcode — APPROVED** [review]
Thanks!

**Review by jvandort — APPROVED** [review]

File: subprojects/docs/src/docs/userguide/getting-started/tutorial/part4_gradle_plugins.adoc [hunk 1]
```diff
@@ -173,7 +174,7 @@ BUILD SUCCESSFUL in 331ms
 
 The `publishToMavenLocal` task builds the POM file and the artifacts to be published. It then _installs_ them into the local Maven repository.
 
-You can view the POM and JAR file in the `build` directory:
+You can view the POM and JSON file in the `build` directory:
```
Comment (jvandort) [inline subprojects/docs/src/docs/userguide/getting-started/tutorial/part4_gradle_plugins.adoc:177]: Generally we call module metadata files `GMM`. It is technically in JSON format, but calling module metadata files `JSON` here is like calling POM files `XML`

Comment (lkasso) [reply subprojects/docs/src/docs/userguide/getting-started/tutorial/part4_gradle_plugins.adoc:177]: Thank you @jvandort!

**Comment by lkasso** [comment]
@bot-gradle test and merge

**Pull request merged**